### PR TITLE
Shard id parameter for system account

### PR DESCRIPTION
**Pull request by miiu96**
- An additional parameter called `forced-shard-id` has been introduced to the `address` group.
- This new parameter makes it easier to retrieve information related to a system account. 
-  Please note that this parameter is `only` applicable when the given address corresponds to a system account.

### CODE REVIEW

File: api/groups/baseAccountsGroup.go [hunk 1]
```diff
@@ -137,7 +137,7 @@ func (group *accountsGroup) getAccounts(c *gin.Context) {
 		return
 	}
 
-	options, err := parseAccountQueryOptions(c)
+	options, err := parseAccountQueryOptions(c, addresses[0])
```
Comment (iulianpascalau) [inline api/groups/baseAccountsGroup.go:140]: I think this can panic if we did not provide any addresses

File: api/groups/errors.go [hunk 2]
```diff
@@ -13,3 +13,6 @@ var ErrHandlerDoesNotExist = errors.New("handler does not exist")
 
 // ErrWrongTypeAssertion signals that a wrong type assertion issue was found during the execution
 var ErrWrongTypeAssertion = errors.New("wrong type assertion")
+
+// ErrShardIDCannotBeProvided signals that the shard id cannot be provided for a different address other than the system account address
+var ErrShardIDCannotBeProvided = errors.New("shard id parameter can be provided only for system account")
```
Comment (bogdan-rosianu) [inline api/groups/errors.go:18]: shard id parameter can only be provided for system accounts

File: api/groups/urlParams.go [hunk 3]
```diff
@@ -9,6 +9,9 @@ import (
 	"github.com/multiversx/mx-chain-proxy-go/common"
 )
 
+// SystemAccountAddressBech is the const for the system account address
+const SystemAccountAddressBech = "erd1lllllllllllllllllllllllllllllllllllllllllllllllllllsckry7t"
```
Comment (bogdan-rosianu) [inline api/groups/urlParams.go:13]: are we sure there aren't other contracts as well? such as the ones that hold 'burn for all' roles

**Review by iulianpascalau — APPROVED** [review]

File: process/accountProcessor.go [hunk 4]
```diff
@@ -513,6 +513,10 @@ func (ap *AccountProcessor) getObserversForAddress(address string, availability
 		return nil, err
 	}
 
+	if optionalShardID.HasValue {
```
Comment (bogdan-rosianu) [inline process/accountProcessor.go:516]: no need to compute the shard id if the value is set
However, you only protected against bad usages of this flag at API level. If one uses proxy as a library and directly call these processors, they might end up with inconsistent results. a simple solution would be renaming shardID to forcedShardID (or something similar) in this case

Comment (miiu96) [reply process/accountProcessor.go:516]: refactored

File: api/groups/urlParams.go (hunk 3 above)
Comment (miiu96) [reply api/groups/urlParams.go:13]: the address for the global settings of esdts is different for every shard.

File: api/groups/errors.go (hunk 2 above)
Comment (miiu96) [reply api/groups/errors.go:18]: changed

File: api/groups/baseAccountsGroup.go (hunk 1 above)
Comment (miiu96) [reply api/groups/baseAccountsGroup.go:140]: added extra check

File: process/accountProcessor.go [hunk 5]
```diff
@@ -502,13 +502,18 @@ func (ap *AccountProcessor) getShardIfOdAddress(address string) (uint32, error)
...
 	if err != nil {
 		return nil, err
 	}
 
-	shardID, err := ap.proc.ComputeShardId(addressBytes)
+	var shardID uint32
+	if forcedShardID.HasValue {
+		shardID = forcedShardID.Value
+	} else {
+		shardID, err = ap.proc.ComputeShardId(addressBytes)
```
Comment (bogdan-rosianu) [inline process/accountProcessor.go:515]: addressBytes can be computed here. in case of a forced shard id, there's no need to compute it

Comment (miiu96) [reply process/accountProcessor.go]: fixed

**Review by bogdan-rosianu — APPROVED** [review]

**Pull request merged**